corehq/apps/cloudcare/views.py

```diff
@@ -269,8 +269,9 @@ def get(self, request, domain, app_id, **kwargs):
         if not app_access.user_can_access_app(request.couch_user, app):
             raise Http404()
 
-        if not request.couch_user.can_access_web_app(domain, app.origin_id):
-            raise Http404()
+        if request.couch_user.is_web_user() or request.couch_user.can_access_any_web_apps(domain):
```

Comment:
In the case of https://github.com/dimagi/commcare-hq/pull/34535 this condition gates filtering the result of `get_web_apps_available_to_user` by `user.can_access_web_app(domain, app_id)`, and here it's gating `couch_user.can_access_web_app(domain, app_id)` which potentially results in 404. So in both cases when this condition is true, user permissions are constrained by `user.can_access_web_app(domain, app_id)`. Thanks for reading while I process on the page.